sdk/go.mod

```diff
@@ -9,8 +9,6 @@ go 1.22.1
 
 toolchain go1.22.5
 
-replace github.com/openbao/openbao/api/v2 => ../api
```

Comment:
While this makes consumption of the sdk easier, it does have the unfortunate side-effect that changes in the API aren't immediately reflected in local testing of the SDK... But presumably the combination of API+SDK functionality is adequately tested through the use in the main module, which does have the correct replace directives to make this work.

Thoughts?

Reply:
I've reverted this change for the time being; I see Brave was able to pull the dependency successfully without it. I've asked on https://github.com/brave-intl/bat-go/pull/2707.